chsdi/models/vector/__init__.py

```diff
@@ -179,7 +180,7 @@ def label_column(cls):
 
     @classmethod
     def geom_filter(cls, geometry, imageDisplay, mapExtent, tolerance, srid):
-        tolerance_meters = get_tolerance_meters(imageDisplay, mapExtent, tolerance)
+        tolerance_meters = get_tolerance_meters(imageDisplay, mapExtent, tolerance, srid)
         geom_column = cls.geometry_column()
         wkb_geometry = WKBElement(buffer(geometry.wkb), srid)
         return func.ST_DWITHIN(geom_column,
```

Comment:
From the docu https://postgis.net/docs/ST_DWithin.html, the distance is in units of srid for geometry not in meter. Not sure here if `transform_geometry(geom_column, wkb_geometry, srid)` returns a geometry or a geography, neither I know what is the difference for postgresql between geometry and geography ?

Reply:
yes the interaction with postgis is always in lv95
```
WHERE ST_DWithin(bfs.gwr_chsdi.the_geom, ST_Transform(ST_GeomFromWKB(%(ST_GeomFromWKB_1)s, %(ST_GeomFromWKB_2)s), %(ST_Transform_1)s), %(ST_DWithin_1)s)
 LIMIT %(param_1)s
2022-07-13 11:30:43,421 INFO  [sqlalchemy.engine.base.Engine][waitress] {'ST_DWithin_1': 4.278382581649971, 'ST_GeomFromWKB_1': <read-only buffer for 0x7f42da81a870, size -1, offset 0 at 0x7f42da81ab30>, 'param_1': 201, 'ST_Transform_1': 2056, 'ST_GeomFromWKB_2': 4326}
```
the input geometry (WGS84) will be transformed to lv95 before it will be used for the st_dwithin function call in postgis:

https://github.com/geoadmin/mf-chsdi3/blob/master/chsdi/models/vector/__init__.py#L185

since the postgis interaction is always in lv95 we can always convert the tolerance into meters. I think we can even skip this transformation for LV95,LV03 and WebMerc which is already in meters https://epsg.io/3857.

Reply:
But `DEFAULT_OUPUT_SRID` is LV03 and not LV95 ? If I understand correctly we have to differentiate between the default SRID of the DB which is LV95, probably set here https://github.com/geoadmin/mf-chsdi3/blob/191da9d9f2129e1c05cf744af3a756c4b894bfa3/chsdi/models/vector/__init__.py#L21 ? and the default output SRID which is LV03. So if this is correct I suggest to create a new variable; `DEFAULT_DB_SRID` set to LV95 and use this in the model https://github.com/geoadmin/mf-chsdi3/blob/191da9d9f2129e1c05cf744af3a756c4b894bfa3/chsdi/models/vector/__init__.py#L21 and also in the `get_tolerance_meters()` method. However in this method I would not use it as parameter simply hardcoded in the transform.

```python
bounds = trans_mapExtent.bounds
# if srs is geographic, convert bounds to meters
if srid == 4326:
    bounds = transform_round_geometry(mapExtent, srid, DEFAULT_DB_SRID, rounding=True).bounds            
```

I agree that if the tolerance is in meter, we only need/should transform WGS84.